cmd/bootkube/render.go

```diff
@@ -196,3 +217,28 @@ func altNamesFromURLs(urls []*url.URL) *tlsutil.AltNames {
 	}
 	return &an
 }
+
+func pickDNSServiceIP(serviceCIDR string) (string, error) {
+	ip, ipnet, err := net.ParseCIDR(serviceCIDR)
+	if err != nil {
+		return "", err
+	}
+	// pick the 10th IP as a conventional default
+	for i := 0; i < 10; i++ {
+		incIPv4(ip)
+	}
+
+	if ipnet.Contains(ip) {
+		return ip.String(), nil
+	}
+	return "", fmt.Errorf("DNS service IP not in range %s", ipnet.String())
```

Comment:
I clarified this a bit by mentioning the IP and range, since small vs big might not always be the problem. Example at the top. `Error: DNS service IP 10.3.0.10 is not in 10.3.0.0/30`